### ISSUE TITLE
有关代码的问题

### ISSUE THREAD

**Issue by foreversunx** [albertyang33/FOREST#2]
杨教授您好：
我对这段代码有些问题想问一下：
`# net emb
            if self.rnn_model.use_network:
                nb1 = self.rnn_model.neighbor_sampling(outputs_id[:,-1].long().cpu().numpy(), self.rnn_model.nnl1) #(batch*1)*nnl1
                nb2 = self.rnn_model.neighbor_sampling(nb1.reshape(-1), self.rnn_model.nnl2) #(batch*len*nnl1)*nnl2    
                nf2 = nn.functional.relu(self.rnn_model.gcn2(self.rnn_model.net_emb[nb2,:]).mean(dim=1).view(-1,self.rnn_model.nnl1,self.rnn_model.nhid)) #(batch*len)*nnl1*nhid
                nf1 = nn.functional.relu(self.rnn_model.gcn1(nf2).mean(dim=1).view(input.size(0),self.rnn_model.nhid))
                nfs[:,t,:] = nf1
                net_emb = nfs[:,t-self.rnn_model.mem_size+1:t+1,:].mean(dim=1)

            step_input = self.rnn_model.encoder(outputs_id[:,-1].long())`
outputs_id[:,-1] 将会一直是0，如果我没理解错的话，这样做的原因是什么呢？

**Comment by albertyang33** [comment]
抱歉没有理解你的问题，这一段outputs_id没有被赋值过，不明白“将会一直是0”是什么意思

**Comment by foreversunx** [comment]
outputs_id = Variable(torch.zeros(batch_size, max_len)).cuda() # with start ids
outputs_id在这里被初始化过，那如果每次都取outputs_id[:,-1]的话，那么取到的tensor值都是0

**Comment by albertyang33** [comment]
model.py 222行有赋值

**Comment by foreversunx** [comment]
是的，但model.py 207行的if语句表明它只是被赋值了start_size列，但却初始化了max_len列，不知道我理解的对不对

**Comment by yangchengbupt** [comment]
有道理，我fix了line 228 235 246三处。这部分主要影响宏观预测任务中采样模拟出的user的邻居编码，考虑到采样本身较大的variance，估计对评测指标影响不大。感谢指出问题，有问题欢迎随时交流！

**Comment by XgSuen** [comment]
谢谢老师的耐心解答！
